reckoner/helm/client.py

```diff
@@ -0,0 +1,159 @@
+from provider import HelmProvider
+from command import HelmCommand
+from reckoner.command_line_caller import Response
+import re
+import logging
+
+
+class HelmClient(object):
+    version_regex = re.compile(r'[a-zA-Z]+: v([0-9\.]+)(\+g[0-9a-f]+)?')
+    repository_header_regex = re.compile(r'^NAME\s+URL$')
+    global_helm_flags = ['debug', 'home', 'host', 'kube-context', 'kubeconfig',
+                         'tiller-connection_timeout', 'tiller-namespace']
+
+    def __init__(self, default_helm_arguments=[], provider=HelmProvider):
+        self._default_helm_arguments = default_helm_arguments
+        self._provider = provider
+
+    @property
+    def default_helm_arguments(self):
+        """The default helm arguments for all commands run through the client."""
+        return self._default_helm_arguments
+
+    @default_helm_arguments.setter
+    def default_helm_arguments(self, value):
+        """Setter of the default helm arguments to override"""
+        self._default_helm_arguments = value
+
+    def execute(self, command, arguments=[], filter_non_global_flags=False):
+        """
+        Run the command with the help of the provider.
+
+        return HelmCmdResponse
+        """
+
+        default_args = list(self.default_helm_arguments)
```

Comment:
I'm getting tripped up here when I don't pass any helm_args:
```
[ do-sfo2-sudermanjr   ] [pw] master  13s
 1 ✘± reckoner --log-level=DEBUG  plot course.yaml --dry-run
2019-01-03 15:05:57 root[83451]    DEBUG helm help
2019-01-03 15:05:57 root[83451]    DEBUG helm version --short --server
2019-01-03 15:05:58 root[83451]    DEBUG Installing repository: {'url': 'https://kubernetes-charts-incubator.storage.googleapis.com', 'name': 'incubator'}
Traceback (most recent call last):
  File "/Users/asuderma/.pentagon/pw/bin/reckoner", line 11, in <module>
    load_entry_point('reckoner', 'console_scripts', 'reckoner')()
  File "/Users/asuderma/.pentagon/pw/lib/python2.7/site-packages/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/Users/asuderma/.pentagon/pw/lib/python2.7/site-packages/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/Users/asuderma/.pentagon/pw/lib/python2.7/site-packages/click/core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/asuderma/.pentagon/pw/lib/python2.7/site-packages/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/asuderma/.pentagon/pw/lib/python2.7/site-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/Users/asuderma/.pentagon/pw/lib/python2.7/site-packages/click/decorators.py", line 17, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/Users/asuderma/repos/reactiveops/reckoner/reckoner/cli.py", line 53, in plot
    h = Reckoner(file=file, dryrun=dry_run, debug=debug, helm_args=helm_args, local_development=local_development)
  File "/Users/asuderma/repos/reactiveops/reckoner/reckoner/reckoner.py", line 72, in __init__
    self.course = Course(file)
  File "/Users/asuderma/repos/reactiveops/reckoner/reckoner/course.py", line 72, in __init__
    repo.install()
  File "/Users/asuderma/repos/reactiveops/reckoner/reckoner/repository.py", line 75, in install
    if self.name not in self._helm_client.repositories:
  File "/Users/asuderma/repos/reactiveops/reckoner/reckoner/helm/client.py", line 64, in repositories
    raw_repositories = self.execute('repo', ['list'], filter_non_global_flags=True).stdout
  File "/Users/asuderma/repos/reactiveops/reckoner/reckoner/helm/client.py", line 35, in execute
    default_args = list(self.default_helm_arguments)
TypeError: 'NoneType' object is not iterable
```

Reply:
looking into it

Reply:
Found the problem! Thanks for tripping over it - I was exclusively testing with `test_course.yml` and not a very basic implementation. One thing for the list is end to end testing with simple and complex course.yml files.

Reply:
Awesome.  Works now.